src/lib/auth/refresh-token-set.ts

```diff
@@ -0,0 +1,46 @@
+// The response shape from
+// POST {IDENTITY_PROVIDER}/oauth2/token
+interface TokenSet {
+	access_token: string
+	refresh_token: string
+	scope: string
+	expires_in: number
+	id_token: string
+	token_type: 'bearer'
+}
+
+/**
+ * Takes a refresh token, and returns a refreshed tokenSet.
+ *
+ * Adapted from: https://next-auth.js.org/tutorials/refresh-token-rotation#server-side
+ */
+export default async function refreshTokenSet(
```

Comment:
💭 Are we able to abstract this a bit to work for any auth provider? Maybe we can pass the current provider's configuration object alongside the `refreshToken`?

Reply:
After looking at this for a bit, it feels like a premature abstraction — going to hold on making this change

Reply:
Right on! Thanks for trying it out anyways 😊